backend/src/main/java/com/tradomator/service/WalletService.java

```diff
@@ -0,0 +1,40 @@
+package com.tradomator.service;
+
+import com.tradomator.model.Wallet;
+import com.tradomator.repo.WalletRepo;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@Service
+public class WalletService {
+
+    private final WalletRepo walletRepo;
+
+
+    public WalletService(WalletRepo walletRepo) {
+        this.walletRepo = walletRepo;
+
+    }
+
+
+    public Wallet getLatestWalletDocument() {
+        List<Wallet> all = walletRepo.findAll();
```

Comment:
```suggestion
      List<Wallet> all = findAllByOrderByUpdateTimeDesc();
      if(all.size() > 0) return all.get(0);
throw new ...
```

Reply:
nice one, was just thinking of a nice exception. Happy to find it here as a next comment :-) like it. thanks cheers t

Reply:
just changed List<Wallet> into Wallet type meanwhile, let's see how to adapt..

Reply:
resolve with: if(latestWallet.getId().isBlank())...